Ensure basic auth honored for healthcheck

jangie opened the pull request:
Related to #274 , in the case of Marathon being protected by Basic authentication, a Marathon host that is marked down would have never been marked back up once down. This specifically caused issue for my team as a dependent project (Traefik) marked a Marathon host as down, thus causing new Marathon deployments to not be monitored by that dependent project.

Explicitly removes `net/http` from `cluster`, and has it instead utilize a new struct shared by `client` that ensures normalization of request handling.

Also fixes a related potential issue if `/ping` is protected by a DCOS token.

timoreimann commented on client_test.go:
```diff
@@ -39,6 +39,14 @@ func TestNewClient(t *testing.T) {
 	assert.Equal(t, conf.PollingWaitTime, defaultPollingWaitTime)
 }
 
+func TestEmptyEndpoint(t *testing.T) {
```
Let's call this `TestInvalidConfig` because that's what we actually care about when verifying the client.

(Specific valid and invalid URLs are covered by `cluster_test.go`.)

jangie commented:
@timoreimann Thanks for the comments! I've addressed them all.

I've also gone ahead and removed the existing `httpClient` property within the `client` struct, as personally I found it potentially confusing that there was both a property as well as a struct called that after addressing the comments. I have modified `subscription.go` as a result. If this is not to your liking, I can add it back and revert my change to `subscription.go`.

Thanks!

jangie commented:
@timoreimann Sorry about that! I should've caught that with my last go-round. Should be ready for you to take a look at again.

jangie commented:
I likewise don't have a DC/OS cluster to test against, and I wasn't yet able to reproduce the issue with basic credentials; we're likely taking auth off of our marathon setup shortly to reduce issues til this is merged. I can at least say that without the credentials, against a basic auth protected marathon environment, that I receive a 401 against `/ping` without credentials, so this should be a step in the right direction imo.

timoreimann commented:
No worries, let's just merge then. If you get a chance to test this eventually, maybe you can leave a quick note here.

Chances are we're going to bump go-marathon in Traefik soon anyway because of containous/traefik#1510; when doing so, I can piggyback this PR if you like, but also feel free to submit a dependency update yourself in Traefik if this is rather urgent for you.

jangie commented:
👍  thanks! i'm curious about how to bump versions of deps and so will attempt to do so just to scratch that itch.
